source/loader/layers/validation/backtrace_lin.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#include "backtrace.hpp"
+
+#include <execinfo.h>
+#include <vector>
+
+namespace validation_layer {
+
+std::vector<BacktraceLine> getCurrentBacktrace() {
+    void *backtraceFrames[MAX_BACKTRACE_FRAMES];
+    int frameCount = backtrace(backtraceFrames, MAX_BACKTRACE_FRAMES);
+    char **backtraceStr = backtrace_symbols(backtraceFrames, frameCount);
+
+    if (backtraceStr == nullptr) {
+        return std::vector<BacktraceLine>(1, "Failed to acquire a backtrace");
+    }
+
+    std::vector<BacktraceLine> backtrace;
+    for (int i = 0; i < frameCount; i++) {
+        backtrace.push_back(BacktraceLine(backtraceStr[i]));
+    }
+
+    free(backtraceStr);
```

Comment:
nit: there's still a (small) chance for a leak here: if push_back or std::string ctor throws, free() is never called.

Reply:
I've changed `push_back` to `emplace_back`, supposedly it's more efficient. The `std::bad_alloc` seems to be the only possible exception in this case. I've put the `emplace_back` inside a try-catch block to handle this exception.